### ISSUE TITLE
confusion in setting steps_per_epoch and validation_steps for model.fit_generator

### ISSUE THREAD

**Issue by shabu19** [keras-team/keras#14595]
hi, I'm using wushidonguc/two-stream-action-recognition-keras code on UCF-101 Dataset.
The dataset is too large, so we are using generators to pass to model.fit_generator function 
I have attached data_list, please have a look at it first.

Now I'm confused on how to set the steps_per_epoch and validation_steps parameters in the fit_generator function. 
The author has used this formula for steps_per_epoch
    steps_per_epoch = (len(data.data_list) * 0.7) / batch_size

the issue is that the length of data_list will simply return the number of subjects and not the number of actual samples of classes
e.g. if i have to train the model for 101 classes (by looking at the data_list) the length of data_list will be 13192. but in reality number of samples are sum of samples of all subject (simply addition of column D of data_list). 
for validation_steps, 1 is being passed to it.

below is the whole code 

`
from tensorflow.keras.callbacks import TensorBoard, ModelCheckpoint, EarlyStopping, CSVLogger, LearningRateScheduler
from temporal_train_model import ResearchModels
from temporal_train_data import DataSet
import time
import os.path
from os import makedirs
from tensorflow.keras.applications import inception_resnet_v2

def train(num_of_snip=5, opt_flow_len=10, saved_model=None, class_limit=None, image_shape=(224, 224), load_to_memory=False, batch_size=32, nb_epoch=100, name_str=None):

    # Learning rate schedule.
    lr_schedule = LearningRateScheduler(fixed_schedule, verbose=0)
 
    # Get the data and process it.
    data = DataSet(
            num_of_snip=num_of_snip,
            opt_flow_len=opt_flow_len,
            image_shape=image_shape,
            class_limit=class_limit,
                )

    # Get samples per epoch.
    # Multiply by 0.7 to attempt to guess how much of data.data is the train set.
    steps_per_epoch = (len(data.data_list) * 0.7) // batch_size

    # Get the data.
    generator = data.stack_generator(batch_size, 'train')
    val_generator = data.stack_generator(batch_size, 'test')

    # Get the model.
    temporal_cnn = ResearchModels(nb_classes=len(data.classes), num_of_snip=num_of_snip, opt_flow_len=opt_flow_len, image_shape=image_shape, saved_model=saved_model)

    # Fit!
    temporal_cnn.model.fit_generator(generator=generator, steps_per_epoch=steps_per_epoch, epochs=nb_epoch, 

validation_data=val_generator, validation_steps=1,0)

def main():
    class_limit = 101         # int, can be 1-101 or None
    num_of_snip = 1         # number of chunks used for each video
    opt_flow_len = 10       # number of optical flow frames used
    image_shape=(224, 224)
    load_to_memory = False  # pre-load the sequences into memory
    batch_size = 32
    nb_epoch = 100

    train(num_of_snip=num_of_snip, opt_flow_len=opt_flow_len, saved_model=saved_model,
            class_limit=class_limit, image_shape=image_shape,
            load_to_memory=load_to_memory, batch_size=batch_size,
            nb_epoch=nb_epoch, name_str=name_str)

if __name__ == '__main__':
    main()
`
[data_list.xlsx](https://github.com/keras-team/keras/files/6315928/data_list.xlsx)
